src/System.Numerics.Experimental/Half.cs

```diff
@@ -341,17 +353,317 @@ public override int GetHashCode()
 
         public override string ToString()
         {
-            return ToString(format: null, formatProvider: null);
+            return $"0x{m_value:X4}";
+            // return ToString(format: null, formatProvider: null);
+            // TODO: Implement this
         }
 
         public string ToString(string format = null, IFormatProvider formatProvider = null)
         {
-            throw new NotImplementedException();
+            return $"0x{m_value:X4}";
+            // throw new NotImplementedException();
+            // TODO: Implement this
         }
 
         public bool TryFormat(Span<char> destination, out int charsWritten, ReadOnlySpan<char> format, IFormatProvider formatProvider)
         {
             throw new NotImplementedException();
         }
+
+        // -----------------------Start of to-half conversions-------------------------
+
+        public static implicit operator Half(int value)
+        {
+            bool sign = value < 0;
+            Half h = (uint)(sign ? -value : value); // Math.Abs but doesn't throw exception, because we cast it to uint anyway
+            return sign ? new Half((ushort)(h.m_value | SignMask)) : h;
+        }
+
+        public static implicit operator Half(uint value)
+        {
+            int shiftDist = BitOperations.LeadingZeroCount(value) - 21;
+            if (shiftDist >= 0)
+                return value != 0?
+                    new Half(false, (ushort)(0x18 - shiftDist), (ushort) (value << shiftDist)):
```

Comment:
nit: The generaly codestyle is no space after the closing parentheses

Reply:
same goes below for cases like `return (int) value`

Reply:
Ah, yes. R# had some weird default settings on... thought I've got rid of all of it.